bins/src/bin/datadog-static-analyzer-server.rs

```diff
@@ -58,13 +58,25 @@ async fn serve_static(
     server_configuration: &State<ServerConfiguration>,
     name: &str,
 ) -> Option<NamedFile> {
-    let name_optional = server_configuration.files_to_serve.get(name);
+    server_configuration.static_directory.as_ref()?;
 
-    if let Some(name) = name_optional {
-        NamedFile::open(name).await.ok()
-    } else {
-        None
+    if name.contains("..") {
+        return None;
     }
+
+    if name.starts_with('.') {
+        return None;
+    }
+
+    let full_path = Path::new(
+        server_configuration
+            .static_directory
+            .clone()
+            .unwrap()
+            .as_str(),
+    )
+    .join(Path::new(name));
+    NamedFile::open(full_path).await.ok()
```

Comment:
What about applying this instead of the current code:
```rust
if server_configuration.static_directory.is_none()
    || name.contains("..")
    || name.starts_with('.')
{
    return None;
}

let s = server_configuration.static_directory.as_ref().unwrap();

let full_path = Path::new(s).join(name);
NamedFile::open(full_path).await.ok()
```

Reply:
I have nothing against this suggestion. Applied it!